src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala

```diff
@@ -123,17 +116,23 @@ class ArticlesRepository(quill: SqliteZioJdbcContext[SnakeCase], dataSource: Dat
   ).unit
     .provide(dsLayer)
 
-  def updateBySlug(updateData: ArticleData, slug: String): IO[Exception, Unit] = run(
-    queryArticle
-      .filter(_.slug == lift(slug.toLowerCase()))
-      .update(
-        record => record.slug -> lift(updateData.slug),
-        record => record.title -> lift(updateData.title),
-        record => record.description -> lift(updateData.description),
-        record => record.body -> lift(updateData.body)
-      )
-  ).unit
-    .mapError(_ => Exceptions.AlreadyInUse("Article name already exists"))
+  def updateBySlug(updateData: ArticleData, slug: String): IO[Exception, Unit] = (for {
+    _ <- run(
+      queryArticle
+        .filter(_.slug == lift(slug.toLowerCase()))
+        .update(
+          record => record.slug -> lift(updateData.slug),
+          record => record.title -> lift(updateData.title),
+          record => record.description -> lift(updateData.description),
+          record => record.body -> lift(updateData.body)
+        )
+    ).mapError(_ => Exceptions.AlreadyInUse("Article name already exists"))
```

Comment:
Can we make this mapping decision based on SQL error code? Otherwise, there is a risk we might catch a different kind of exception and claim that's because of article name conflict.